From efa864096449f6f22ad84e825f24d854f0781216 Mon Sep 17 00:00:00 2001
From: Ky-Anh Huynh <kyanh@theslinux.org>
Date: Tue, 16 Apr 2013 21:47:16 +0000
Subject: [PATCH 1/3] Allow some package information to be fetched externally

The original "makepkg" will reset all package information, and that diasllows
to support some hooks (no we do not have any hook when using PKGBUILD.)
---
 pacman/src/pacman-4.1.0/scripts/makepkg.sh.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in b/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
index da620a4..389dee7 100644
--- a/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
+++ b/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
@@ -2733,6 +2733,11 @@ unset pkgname pkgbase pkgver pkgrel epoch pkgdesc url license groups provides
 unset md5sums replaces depends conflicts backup source install changelog build
 unset makedepends optdepends options noextract
 
+pkgname="${PACKAGE_NAME:-}"
+pkgver="${PACKAGE_VERSION:-}"
+pkgrel="${PACKAGE_RELEASE:-}"
+pkgbase="${PACKAGE_BASE:-}"
+
 BUILDFILE=${BUILDFILE:-$BUILDSCRIPT}
 if [[ ! -f $BUILDFILE ]]; then
 	if [[ -t 0 ]]; then
-- 
1.8.2.1


From e01c9a40f396348573a4a709032dc1f7a25a92ed Mon Sep 17 00:00:00 2001
From: Ky-Anh Huynh <kyanh@theslinux.org>
Date: Sat, 20 Apr 2013 16:55:36 +0700
Subject: [PATCH 2/3] Accept checksum (512) from environment

---
 pacman/src/pacman-4.1.0/scripts/makepkg.sh.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in b/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
index 389dee7..b887819 100644
--- a/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
+++ b/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
@@ -2738,6 +2738,10 @@ pkgver="${PACKAGE_VERSION:-}"
 pkgrel="${PACKAGE_RELEASE:-}"
 pkgbase="${PACKAGE_BASE:-}"
 
+if [[ -n "${PACKAGE_SHA512SUMS}" ]]; then
+	sha512sums=("${PACKAGE_SHA512SUMS[@]}")
+fi
+
 BUILDFILE=${BUILDFILE:-$BUILDSCRIPT}
 if [[ ! -f $BUILDFILE ]]; then
 	if [[ -t 0 ]]; then
-- 
1.8.2.1


From ba7d6b9371e49a24631d135d31fa44bbb2c0f66b Mon Sep 17 00:00:00 2001
From: Ky-Anh Huynh <kyanh@theslinux.org>
Date: Sat, 20 Apr 2013 17:11:18 +0700
Subject: [PATCH 3/3] pkgbase will define the behavior

As "pkgbase" is a string, while "pkgname" may be an array,
we can only get "pkgbase" from environment.
---
 pacman/src/pacman-4.1.0/scripts/makepkg.sh.in | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in b/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
index b887819..b1eb313 100644
--- a/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
+++ b/pacman/src/pacman-4.1.0/scripts/makepkg.sh.in
@@ -2733,7 +2733,6 @@ unset pkgname pkgbase pkgver pkgrel epoch pkgdesc url license groups provides
 unset md5sums replaces depends conflicts backup source install changelog build
 unset makedepends optdepends options noextract
 
-pkgname="${PACKAGE_NAME:-}"
 pkgver="${PACKAGE_VERSION:-}"
 pkgrel="${PACKAGE_RELEASE:-}"
 pkgbase="${PACKAGE_BASE:-}"
@@ -2765,6 +2764,17 @@ else
 	source_safe "$BUILDFILE"
 fi
 
+# There are some cases
+#   PACKAGE_BASE is unset:
+#     `pkgname`, `pkgbase` will be from `PKGBUILD`
+#
+#   PACKAGE_BASE is set:
+#     `pkgbase` is defined
+#     `pkgname` will be `pkgbase` if `pkgname` is unset
+#
+if [[ -z "$pkgname" ]]; then
+	pkgname="$pkgbase"
+fi
 # set defaults if they weren't specified in buildfile
 pkgbase=${pkgbase:-${pkgname[0]}}
 epoch=${epoch:-0}
-- 
1.8.2.1

